packages/hint-doctype/src/hint.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * @fileoverview Hint to validate if the doctype is correct
+ */
+
+import * as os from 'os';
+import { Category } from 'hint/dist/src/lib/enums/category';
+import { HintScope } from 'hint/dist/src/lib/enums/hintscope';
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { IHint, HintMetadata, FetchEnd, ProblemLocation } from 'hint/dist/src/lib/types';
+import { MatchInformation } from './types';
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class implements IHint {
+
+    public static readonly meta: HintMetadata = {
+        docs: {
+            category: Category.interoperability,
+            description: `This hint checks if the HTML is using the most modern DOCTYPE.`
+        },
+        id: 'doctype',
+        schema: [],
+        scope: HintScope.any
+    }
+
+    public constructor(context: HintContext) {
+        const correctLine = 0;
+        const doctypeRegExp = new RegExp(`(<!doctype\\s+(html)\\s*>)(.+)?`, 'gi');
+        const doctypeFlexibleRegExp = new RegExp(`<[^>]*doctype[^<]*>`, 'gi');
+
+        const defaultProblemLocation: ProblemLocation = {
+            column: 0,
+            line: 0
+        };
+
+        const report = async (resource: string, message: string, problemLocation = defaultProblemLocation): Promise<void> => {
+            await context.report(resource, null, message, undefined, problemLocation);
+        };
+
+        const getCurrentDoctypeProblemLocation = (text: string): ProblemLocation[] => {
+            const lines = text.split(os.EOL);
+            const locations: ProblemLocation[] = [];
+
+            lines.forEach((line: string, i: number): void => {
+                const matched = doctypeRegExp.exec(line);
+
+                if (matched){
+                    locations.push({
+                        column: matched.index,
+                        line: i
+                    });
+                }
+            });
+
+            return locations;
+        };
+
+        const getMatchInformation = (text: string): MatchInformation => {
+            return {
+                locations: getCurrentDoctypeProblemLocation(text),
+                matches: text.match(doctypeRegExp)
+            };
+        };
+
+        const checkNoDoctypeInContent = async (matchInfo: MatchInformation, resource: string, content: string): Promise<boolean> => {
+            if (matchInfo.matches && matchInfo.matches.length > 0) {
+                return true;
+            }
+
+            if (!doctypeFlexibleRegExp.exec(content)) {
+                await report(resource, `'DOCTYPE' was not specified.`);
+
+                return false;
+            }
+
+            await report(resource, `'DOCTYPE' should be specified as '<!doctype html>'.`);
```

Comment:
For consistency with other error messages, let' use `doctype` instead of `DOCTYPE` here and everywhere else.